server/sonar-server/src/main/java/org/sonar/server/projectbranch/ws/ShowAction.java

```diff
@@ -0,0 +1,167 @@
+
+public class ShowAction implements BranchWsAction {
+
+  private final DbClient dbClient;
+  private final UserSession userSession;
+  private final ComponentFinder componentFinder;
+
+  public ShowAction(DbClient dbClient, UserSession userSession, ComponentFinder componentFinder) {
+    this.dbClient = dbClient;
+    this.userSession = userSession;
+    this.componentFinder = componentFinder;
+  }
+
+  @Override
+  public void define(WebService.NewController context) {
+    WebService.NewAction action = context.createAction(ACTION_SHOW)
+      .setSince("6.6")
+      .setDescription("Show branch information of a project")
+      .setResponseExample(Resources.getResource(getClass(), "show-example.json"))
+      .setInternal(true)
+      .setHandler(this);
+
+    action
+      .createParam(PARAM_COMPONENT)
+      .setDescription("Component key")
+      .setExampleValue(KEY_FILE_EXAMPLE_001)
+      .setRequired(true);
+
+    action
+      .createParam(PARAM_BRANCH)
+      .setDescription("Branch key")
+      .setExampleValue(KEY_BRANCH_EXAMPLE_001)
+      .setRequired(true);
+  }
+
+  @Override
+  public void handle(Request request, Response response) throws Exception {
+    String projectKey = request.mandatoryParam(PARAM_COMPONENT);
+    String branchName = request.mandatoryParam(PARAM_BRANCH);
+
+    try (DbSession dbSession = dbClient.openSession(false)) {
+      ComponentDto component = componentFinder.getByKeyAndBranch(dbSession, projectKey, branchName);
+      userSession.checkComponentPermission(UserRole.USER, component);
+      ComponentDto project = componentFinder.getByUuid(dbSession, component.projectUuid());
+
+      List<MetricDto> metrics = dbClient.metricDao().selectByKeys(dbSession, asList(ALERT_STATUS_KEY, BUGS_KEY, VULNERABILITIES_KEY, CODE_SMELLS_KEY));
+      Map<Integer, MetricDto> metricsById = metrics.stream().collect(uniqueIndex(MetricDto::getId));
+      Map<String, Integer> metricIdsByKey = metrics.stream().collect(uniqueIndex(MetricDto::getKey, MetricDto::getId));
+
+      BranchDto branch = getBranch(dbSession, component.projectUuid());
+      String mergeBranchUuid = branch.getMergeBranchUuid();
+      BranchDto mergeBranch = mergeBranchUuid == null ? null : getBranch(dbSession, mergeBranchUuid);
+
+      Multimap<String, MeasureDto> measuresByComponentUuids = dbClient.measureDao()
+        .selectByComponentsAndMetrics(dbSession, Collections.singletonList(branch.getUuid()), metricsById.keySet())
+        .stream().collect(index(MeasureDto::getComponentUuid));
+
+      WsUtils.writeProtobuf(buildResponse(branch, project, mergeBranch, metricIdsByKey, measuresByComponentUuids), request, response);
+    }
+  }
+
+  private BranchDto getBranch(DbSession dbSession, String uuid) {
+    Optional<BranchDto> branch = dbClient.branchDao().selectByUuid(dbSession, uuid);
+    checkState(branch != null, "Branch uuid '%s' not found", uuid);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this null-check of an "Optional". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2789)